workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerCreation/SendCreatePlayerRequestSystem.cs

```diff
@@ -29,48 +42,116 @@ protected override void OnCreateManager()
+            ));
+            playerRequestQueued = false;
         }
 
         private void RetryCreatePlayerRequest()
         {
-            RequestPlayerCreation(serializedArgumentsCache);
+            if (playerCreationAttempts < PlayerLifecycleConfig.MaxPlayerCreationAttempts)
+            {
+                ++playerCreationAttempts;
+
+                logDispatcher.HandleLog(LogType.Warning, new LogEvent(
+                    $"Retrying player creation request, attempt {playerCreationAttempts}."
+                ));
+
+                SendCreatePlayerRequest();
+            }
+            else
+            {
+                logDispatcher.HandleLog(LogType.Error, new LogEvent(
+                    $"Unable to create player after {playerCreationAttempts} attempts."
+                ));
+            }
         }
 
         protected override void OnUpdate()
         {
-            if (PlayerLifecycleConfig.AutoRequestPlayerCreation && !initializationGroup.IsEmptyIgnoreFilter)
+            if (playerCreatorEntityIds.Count > 0)
             {
-                RequestPlayerCreation();
-            }
+                if (playerRequestQueued)
+                {
+                    SendCreatePlayerRequest();
+                }
 
-            // Currently this has a race condition where you can receive two entities
-            // The fix for this is more sophisticated server side handling of requests
-            var responses = commandSystem.GetResponses<PlayerCreator.CreatePlayer.ReceivedResponse>();
+                // Currently this has a race condition where you can receive two entities
+                // The fix for this is more sophisticated server side handling of requests
+                var responses = commandSystem.GetResponses<PlayerCreator.CreatePlayer.ReceivedResponse>();
 
-            for (var i = 0; i < responses.Count; i++)
-            {
-                ref readonly var response = ref responses[i];
-                if (response.StatusCode == StatusCode.AuthorityLost)
+                for (var i = 0; i < responses.Count; i++)
                 {
-                    RetryCreatePlayerRequest();
+                    ref readonly var response = ref responses[i];
+                    if (response.StatusCode == StatusCode.AuthorityLost)
+                    {
+                        RetryCreatePlayerRequest();
+                    }
+                    else if (response.StatusCode != StatusCode.Success)
+                    {
+                        logDispatcher.HandleLog(LogType.Error, new LogEvent(
+                            $"Create player request failed: {response.Message}"
+                        ));
+                    }
                 }
-                else if (response.StatusCode != StatusCode.Success)
+            }
+            else
+            {
+                var entityQueryResponses = commandSystem.GetResponses<WorldCommands.EntityQuery.ReceivedResponse>();
```

Comment:
Small nit we can refactor these so Update is just: 

```csharp
if (playerCreatorEntityIDs.Count >0) 
{
	HandleCreateEntity();
}
else
{
	HandleEntityQueryResponses();
}
```

or similar for legibilities sake